astroid/tests/unittest_python3.py

```diff
@@ -258,7 +258,74 @@ def test_underscores_in_numeral_literal(self):
             self.assertIsInstance(inferred, nodes.Const)
             self.assertEqual(inferred.value, expected)
 
+    @require_version('3.6')
+    def test_async_comprehensions(self):
+        return_async_comp_funcs = [
+            extract_node("async def f(): return [i async for i in aiter() if i % 2]"),
+            extract_node("async def f(): return {i async for i in aiter() if i % 2}"),
+            extract_node("async def f(): return (i async for i in aiter() if i % 2)"),
+            extract_node("async def f(): return {i: i async for i in aiter() if i % 2}")
+        ]
+        return_non_async_comp_funcs = [
+            extract_node("async def f(): return {i: i for i in iter() if i % 2}")
+        ]
+
+        for func in return_async_comp_funcs:
+            comp = func.body[-1].value.generators[0]
+            self.assertTrue(comp.is_async.value)
+        for func in return_non_async_comp_funcs:
+            comp = func.body[-1].value.generators[0]
+            self.assertFalse(comp.is_async.value)
+
+    @require_version('3.7')
+    def test_async_comprehensions_outside_coroutine(self):
+        """
```

Comment:
A comment should be enough here. If I remember correctly, docstrings have special meaning in unittests.